File: Tipos de Dados/Ordinais/ordinais.cpp
//C++
//Em C++, podemos definir enumerações para valores ordinais personalizados, mas subfaixas não são suportadas diretamente.
#include <iostream>

using namespace std;

enum DiaDaSemana { Segunda, Terca, Quarta, Quinta, Sexta, Sabado, Domingo };

int main() {
    DiaDaSemana hoje = Terca;

    // Exibindo valor ordinal
    cout << "Hoje é: " << hoje << endl;

    // Usando a posição ordinal
    cout << "Posição ordinal: " << static_cast<int>(hoje) << endl;

    // Comparação
    if (hoje == Terca) {
        cout << "Hoje é terça-feira." << endl;
    }

    return 0;
}

File: Tipos de Dados/Ordinais/Ordinais.java
//Java
//Java oferece suporte a enumerações, mas não a subfaixas.
public class Ordinais {
    enum DiaDaSemana {
        SEGUNDA, TERCA, QUARTA, QUINTA, SEXTA, SABADO, DOMINGO
    }

    public static void main(String[] args) {
        DiaDaSemana hoje = DiaDaSemana.TERCA;

        // Exibindo valor ordinal
        System.out.println("Hoje é: " + hoje);

        // Usando a posição ordinal
        System.out.println("Posição ordinal: " + hoje.ordinal());

        // Comparação
        if (hoje == DiaDaSemana.TERCA) {
            System.out.println("Hoje é terça-feira.");
        }
    }
}
